pinot-tools/src/main/java/com/linkedin/pinot/tools/FilerSegmentPurger.java

```diff
@@ -0,0 +1,104 @@
+ */
+package com.linkedin.pinot.tools;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.model.IdealState;
+import org.kohsuke.args4j.Option;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+public class FilerSegmentPurger extends AbstractBaseCommand implements Command {
+
+  @Option(name = "-zkAddress", required = true, metaVar = "", usage = "Zookeeper address. comma separated host:port pairs")
+  private String _zkAddress;
+
+  @Option(name = "-clusterName", required = true, metaVar = "<String>", usage = "Pinot cluster name.")
+  private String _clusterName;
+
+  @Option(name = "-segmentMountDir", required = true, metaVar = "<String>", usage = "Pinot segment directory mount path on NFS. Segments for a table must be found at segmentMountDir/<clusterName>/<tableName>")
+  private String _segmentMountDir;
+
+  @Option(name = "-help", required = false, help = true, aliases={"-h", "--h", "--help"}, usage = "Print this message.")
+  private boolean _help = false;
+  
+  @Override
+  public boolean execute() throws Exception {
+    ZKHelixAdmin zkHelixAdmin = new ZKHelixAdmin(_zkAddress);
+    Set<String> tableNamesFromZK = Sets.newHashSet();
+    List<String> tableNames = zkHelixAdmin.getResourcesInCluster(_clusterName);
+    for (String tableName : tableNames) {
+      tableNamesFromZK.add(tableName);
+    }
+
+    Map<String, Set<File>> segmentsToPurgeMap = Maps.newHashMap();
+
+    for (String tableName : tableNamesFromZK) {
+      Set<String> segmentsFromIS = Collections.emptySet();
+      // get segment list from Helix if the table exists
+      if (tableNamesFromZK.contains(tableName)) {
+        IdealState idealState = zkHelixAdmin.getResourceIdealState(_clusterName, tableName);
+        segmentsFromIS = idealState.getPartitionSet();
+      }
+      // get segment list from Filer if the table exists
+      String rawTableName = tableName.replace("_OFFLINE", "").replace("_REALTIME", "");
+      File tableRootDir = new File(new File(_segmentMountDir, _clusterName), rawTableName);
+      Set<File> canBePurged = Sets.newHashSet();
+      if (tableRootDir.exists()) {
+        File[] segmentFiles = tableRootDir.listFiles();
+        for(File segmentFile:segmentFiles){
+          //we might have files with or without extension  
+          String segmentName = segmentFile.getName().replace(".tar.gz", "");
+          if(!segmentsFromIS.contains(segmentName)){
+            canBePurged.add(segmentFile);
+          }
+        }
+      }
+      if (!canBePurged.isEmpty()) {
+        segmentsToPurgeMap.put(tableName, canBePurged);
+      }
+    }
+
+    System.out.println("SEGMENTS THAT CAN BE PURGED FROM FILER");
+    long totalPurgeableSize = 0;
+    for (String tableName : segmentsToPurgeMap.keySet()) {
+      System.out.println(tableName);
```

Comment:
Put table name on each line or drop it so that output can be easily cut-paste for commandline deletion (since segment names may not have tar.gz)